src/providers/cloudstack/network.rs

```diff
@@ -14,7 +14,7 @@ const SERVER_ADDRESS: &str = "SERVER_ADDRESS";
 
 #[derive(Clone, Debug)]
 pub struct CloudstackNetwork {
-    server_address: IpAddr,
+    server_address: String,
```

Comment:
This is an URL now, not an address, so should probably be renamed to `server_base_url` or similar.